test/functional/user_sessions_controller_test.rb

```diff
@@ -323,4 +329,14 @@ class UserSessionsControllerTest < ActionController::TestCase
     assert_redirected_to root_url
     assert_equal flash[:error], I18n.t('user_sessions_controller.user_has_been_moderated', username: user.username).html_safe
   end
+
+  test "user that links provider to existing account should not be redirected to dashboard on oauth signup" do
+    # this omniauth config user has the same email as a user in the database
+    request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:github4]
```

Comment:
Also we will require tests for all the 4 providers. This is just for GitHub

Reply:
No problem! I will add them :+1: